packages/@uppy/companion/src/server/emitter/redis-emitter.js

```diff
@@ -1,15 +1,45 @@
-const NRP = require('node-redis-pubsub')
+const redis = require('redis')
 
 /**
  * This module simulates the builtin events.EventEmitter but with the use of redis.
  * This is useful for when companion is running on multiple instances and events need
  * to be distributed across.
  */
 module.exports = (redisUrl, redisPubSubScope) => {
-  const nrp = new NRP({ url: redisUrl, scope: redisPubSubScope })
+  const prefix = redisPubSubScope ? `${redisPubSubScope}:` : ''
+  const publisher = redis.createClient({ url: redisUrl })
+  let subscriber
+
+  const connectedPromise = publisher.connect().then(() => {
+    subscriber = publisher.duplicate()
+    return subscriber.connect()
+  })
+
+  let errorHandler
 
   function on (eventName, handler) {
-    nrp.on(eventName, handler)
+    if (eventName === 'error') {
+      errorHandler = handler
```

Comment:
won't this overwrite any existing error handler? Maybe we should use a node EventEmitter instead

Reply:
I still think we can do better than that though. Alternatively maybe change the interface so that instead of using `on('error',...)`, `removeListener('error',...)` maybe we can expose a new function `setErrorHandler` (I don't think companion ever registers an error handler at all so it doesn't matter). With the current implementation it looks like it's an `error` event, but in fact it's not.